nginx.conf

```diff
@@ -0,0 +1,81 @@
+events {
+    worker_connections 1024;
+}
+
+# frontend
+http {
+    server {
+        listen 80;
+        server_name sim.genlayer.com;
+        return 301 https://$server_name$request_uri;
+    }
+
+    server {
+        listen 443 ssl;
+        http2 on;
+        server_name sim.genlayer.com;
+
+        ssl_certificate /etc/nginx/ssl/sim.genlayer.com.crt;
+        ssl_certificate_key /etc/nginx/ssl/sim.genlayer.com.key;
+        ssl_client_certificate /etc/nginx/ssl/cloudflare.crt;
+        ssl_verify_client off;
+        ssl_protocols TLSv1.2 TLSv1.3;
+        ssl_prefer_server_ciphers off;
+        ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+
+        # Security headers
+        add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload" always;
+        add_header X-Frame-Options DENY;
+        add_header X-Content-Type-Options nosniff;
+        add_header X-XSS-Protection "1; mode=block";
+
+        # Debugging headers (you can remove these in production)
+        # add_header X-Debug-Message "Reached HTTPS server block" always;
+        add_header X-Forwarded-Proto $scheme always;
+        add_header X-Original-URI $request_uri always;
+
+        location / {
+            proxy_pass http://frontend:8080;
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+
+            # Debugging header (you can remove this in production)
+            # add_header X-Debug-Proxy "Reached proxy location block" always;
+        }
+    }
+
+# backend (jsonrpc)
+    server {
+        listen 8443 ssl;
+        http2 on;
+        server_name sim.genlayer.com;
+
+        ssl_certificate /etc/nginx/ssl/sim.genlayer.com.crt;
+        ssl_certificate_key /etc/nginx/ssl/sim.genlayer.com.key;
+        ssl_client_certificate /etc/nginx/ssl/cloudflare.crt;
```

Comment:
all of these names should also be configurable, or at least explained in the README.md